crates/wallet/src/wallet/mod.rs

```diff
@@ -744,25 +692,53 @@ impl Wallet {
     /// index defined in [BIP32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki),
     /// then returns the last revealed address.
     ///
-    /// # Errors
+    /// **WARNING:** You must persist the staged wallet changes after making this call. For example:
     ///
-    /// If writing to persistent storage fails.
-    pub fn reveal_next_address(&mut self, keychain: KeychainKind) -> anyhow::Result<AddressInfo> {
+    /// ```rust,no_run
+    /// # use bdk_wallet::Wallet;
+    /// # use bdk_wallet::signer::{SignersContainer, SignerOrdering};
+    /// # use bdk_wallet::descriptor::Descriptor;
+    /// # use bitcoin::key::Secp256k1;
+    /// # use bdk_wallet::KeychainKind;
+    /// # use bdk_sqlite::{Store, rusqlite::Connection};
+    /// #
+    /// # fn main() -> Result<(), anyhow::Error> {
+    /// # use bitcoin::Network::Testnet;
+    /// # use bdk_chain::persist::Persist;
+    /// # use bdk_wallet::KeychainKind::Internal;
+    /// # let temp_dir = tempfile::tempdir().expect("must create tempdir");
+    /// # let file_path = temp_dir.path().join("store.db");
+    /// # let conn = Connection::open(file_path).expect("must open connection");
+    /// let mut db = Store::new(conn).expect("must create db");
+    /// # let secp = Secp256k1::new();
+    ///
+    /// # let external_descriptor = Descriptor::parse_descriptor(&secp, "wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/1'/0'/0/*)").unwrap();
+    /// # let internal_descriptor = Descriptor::parse_descriptor(&secp, "wpkh(tprv8ZgxMBicQKsPdy6LMhUtFHAgpocR8GC6QmwMSFpZs7h6Eziw3SpThFfczTDh5rW2krkqffa11UpX3XkeTTB2FvzZKWXqPY54Y6Rq4AQ5R8L/84'/1'/0'/1/*)").unwrap();
+    /// # let changeset = db.load_changes()?.expect("Changeset must exist");
+    /// # let mut wallet = Wallet::load_from_changeset(changeset)?;
+    /// let next_address = wallet.reveal_next_address(KeychainKind::External);
+    /// db.write_changes(&wallet.take_staged())?;
+    /// // Now it's safe to show the user their next address!
+    /// println!("Next address: {}", next_address.address);
+    /// # Ok(())
+    /// # }
+    /// ```
```

Comment:
somewhat minimal but I think it compiles

```rust
/// ```rust,no_run
/// # use bdk_chain::persist::Persist;
/// # use bdk_wallet::wallet::{Wallet, ChangeSet};
/// # use bdk_wallet::KeychainKind;
/// # use bdk_sqlite::{Store, rusqlite::Connection};
/// # let conn = Connection::open_in_memory().expect("must open connection");
/// # let mut db = Store::new(conn).expect("must create store");
/// # let changeset = ChangeSet::default();
/// # let mut wallet = Wallet::load_from_changeset(changeset).expect("load wallet");
/// let next_address = wallet.reveal_next_address(KeychainKind::External);
/// db.write_changes(&wallet.take_staged())?;
///
/// // Now it's safe to show the user their next address!
/// println!("Next address: {}", next_address.address);
/// # Ok::<(), anyhow::Error>(())
/// ```
```